jobspec/parse_group.go

```diff
@@ -274,3 +283,61 @@ func parseRestartPolicy(final **api.RestartPolicy, list *ast.ObjectList) error {
 	*final = &result
 	return nil
 }
+
+func parseVolumes(out *map[string]*api.Volume, list *ast.ObjectList) error {
+	volumes := make(map[string]*api.Volume, len(list.Items))
+
+	for _, item := range list.Items {
+		n := item.Keys[0].Token.Value().(string)
+		valid := []string{
+			"type",
+			"read_only",
+			"hidden",
+			"config",
+		}
+		if err := helper.CheckHCLKeys(item.Val, valid); err != nil {
+			return err
+		}
+
+		var m map[string]interface{}
+		if err := hcl.DecodeObject(&m, item.Val); err != nil {
+			return err
+		}
+
+		// TODO(dani): FIXME: this is gross but we don't have ObjectList.Filter here
```

Comment:
If this is not a bug and it isn't going to be addressed before merging, I'd suggest dropping the `FIXME` label.